base/server/python/pki/server/__init__.py

```diff
@@ -1037,6 +1091,35 @@ def __repr__(self):
             return "Dogtag 9 " + self.name
         return self.name
 
+    def cert_update_config(self, cert_id, cert, subsystem=None):
+        """
+        Update corresponding subsystem's CS.cfg with the new cert details
+        passed.
+
+        **Note:**
+        *subsystem* param is ignored when `(cert_id == sslserver ||
+        cert_id == subsystem)` since these 2 certs are used by all subsystems
```

Comment:
This is ok for now, but ideally I think cert_update_config() should not take a subsystem object since it allows the user to provide the instance with a subsystem that doesn't belong to it and there's no validation.

Ideally I think the method should take a cert_id, or a cert_tag and a subsystem_name:
```
def cert_update_config(self, cert_id, cert):
def cert_update_config(self, cert_tag, cert, subsystem_name=None):
```
Then the method can get the subsystem object internally:
```
subsystem = self.get_subsystem(subsystem_name)
```

Reply:
how about we give the caller the freedom to pass _either the loaded subsystem OR the subsystem name_?

The method definition will become something like:

````
def cert_update_config(self, cert_id, cert, subsystem_name=None, subsystem=None):
     ....
    if subsystem_name:
        subsystem = self.get_subsystem(subsystem_name)
    if subsystem:
        subsystem.update_subsystem_cert(cert)
        subsystem.save()
    else:
        <throw exception>
````